restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/SingleExpressionResolver.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.resolver;
+
+import io.stargate.web.docsapi.service.ExecutionContext;
+import io.stargate.web.docsapi.service.query.FilterExpression;
+import io.stargate.web.docsapi.service.query.search.resolver.filter.impl.InMemoryCandidatesFilter;
+import io.stargate.web.docsapi.service.query.search.resolver.filter.impl.PersistenceCandidatesFilter;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.AllFiltersResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.InMemoryDocumentsResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.PersistenceDocumentsResolver;
+import java.util.Optional;
+
+/** Static resolver that resolves from the single {@link FilterExpression}. */
+public final class SingleExpressionResolver {
+
+  private SingleExpressionResolver() {}
+
+  /**
+   * Returns a document resolver for a single {@link FilterExpression} without a parent resolver.
+   *
+   * @param expression {@link FilterExpression}
+   * @param context {@link ExecutionContext}
+   * @return DocumentsResolver
+   */
+  public static DocumentsResolver resolve(FilterExpression expression, ExecutionContext context) {
+    return resolve(expression, context, null);
+  }
+
+  /**
+   * Returns a document resolver for a single {@link FilterExpression}.
+   *
+   * @param expression {@link FilterExpression}
+   * @param context {@link ExecutionContext}
+   * @param parent parent resolver or <code>null</code>
+   * @return DocumentsResolver
+   */
+  public static DocumentsResolver resolve(
+      FilterExpression expression, ExecutionContext context, DocumentsResolver parent) {
```

Comment:
I commented about this before, but it still feels a bit awkward to me to have this in a separate class. It looks like this is really a sub-case of `CnfResolver.resolve(...)` when the `AND` expression has only one child. Could this case be handled by `CnfResolver`?

Reply:
For example, running this test appear to produce equivalent resolvers:
```
ExecutionContext context = ExecutionContext.create(true);
FilterPath filterPath = ImmutableFilterPath.of(Collections.singletonList("field"));
//BaseCondition condition1 = ImmutableStringCondition.of(GtFilterOperation.of(), "find-me");
BaseCondition condition1 = ImmutableStringCondition.of(NeFilterOperation.of(), "find-me");
FilterExpression expression1 = ImmutableFilterExpression.of(filterPath, condition1);

DocumentsResolver r1 = CnfResolver.resolve(And.of(expression1), context);
DocumentsResolver r2 = SingleExpressionResolver.resolve(expression1, context);
```
WDYT?

Reply:
I am removing this class.

Reply:
Done.

Reply:
Done.